feat: adding RLN support for nwaku-compose (#7)
waku-org/nwaku-compose: README.md

Edit the environment variables present at the beginning of the `docker-compose.yml` file.

> It is required to set the `ETH_CLIENT_ADDRESS` environment variable before starting the instance.

Reviewer: Make sure it is edited within the "docker-compose.yml" file:

```suggestion
It is required to set the `ETH_CLIENT_ADDRESS` environment variable within the `docker-compose.yml` file before starting the instance.
```

Author: I didn't specify it because one line above we instruct to edit the env variables in `docker-compose.yml`, so didn't want to repeat the same point made in the previous line.

What do you think? Is it better to repeat it?

Edit the environment variables present at the beginning of the `docker-compose.yml` file.

Get a node's URL:

It is required to set the `ETH_CLIENT_ADDRESS` environment variable before starting the instance.

`ETH_CLIENT_ADDRESS` must be a WebSockets URL for your Ethereum Node.

> For the current default contract, it must a node for the Sepolia network.

Reviewer: ```suggestion
For the current default contract, it must be a node for the Sepolia network.
```

Author: Oh I missed it, will fix :)

In case you're not running your own node, you can get it from [Infura](https://www.infura.io/)

> You can set the environment variable from inside the `docker-compose.yml` file or you can simply run
> 
> ```console
> export ETH_CLIENT_ADDRESS=<your_eth_client_address_here>
> ```

Reviewer: Better only mention modifying the env var in the docker-compose file, and then, create it as an anchor. Something like:

```
x-eth-client-address: &eth_client_address ${ETH_CLIENT_ADDRESS:-<YOUR_ETH_CLIENT_ADDRESS_HERE>}
```
As an example, check how `*pg_env` is used.

```suggestion
You can set the environment variable from inside the `docker-compose.yml` file.
```

Author: Yes, I think that as an anchor is better! Will set it up as an anchor along with the optional RLN env variables that I'm about to add (`rln-relay-cred-password` and `rln-relay-cred-path`).

I mentioned the option of exporting the variable from the shell so it's more user friendly to some, as they don't need to look for and change any files in order to run their instance.
It was based on @alrevuelta idea of how the user experience should be, but not sure if it's redundant or if it might confuse users to mention two ways of doing the same thing.

* `NWAKU_IMAGE` - the image you want to use for the nwaku container (e.g. `NWAKU_IMAGE=statusteam/nim-waku:v0.19.0-rc.0`)
* `DOMAIN` - domain name pointing to the IP address of your node, when configured the run script will request SSL certs from Let's Encrypt and run Waku node with WebSockets Secure (WSS) options enabled (e.g. `DOMAIN=waku.example.com`)
* `NODEKEY` - this env variable allows you to provide a node key as described in [operators documentation](https://github.com/waku-org/nwaku/blob/master/docs/operators/how-to/configure-key.md) (e.g. `NODEKEY=9f439983aa4851346cfe6e17585e426f482871a43626812e23490895cd602c11`)

> * `RLN_RELAY_CONTRACT_ADDRESS` - address of the RLN Relay Contract. It defaults to `0xF471d71E9b1455bBF4b85d475afb9BB0954A29c4` on Sepolia testnet

Reviewer: Small nitpick, avoid hard-coded values that we may need to update in the future. Is enough to only hard-code in the docker-compose file.

```suggestion
* `RLN_RELAY_CONTRACT_ADDRESS` - address of the RLN Relay Contract. It defaults to a Sepolia testnet address.
```

Author: Done :)
